### ISSUE TITLE
Fix `value-keyword-case` false positives for mixed case `ignoreFunctions` option

### ISSUE THREAD

**Issue by Arkellys** [stylelint/stylelint#6436]
### What steps are needed to reproduce the bug?

Setup the following code:

```scss
$appColors: (
  hoveringMenu: (
    background: red;
  )
);

@function getAppColor($name, $variant...) {
  @return map.get($appColors, $name, $variant...);
}

.hoveringToolbar {
  background-color: getAppColor(hoveringMenu, background);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "value-keyword-case": ["lower", {
      "ignoreProperties": ["$appColors"],
      "ignoreFunctions": ["getAppColor"],
    }]
  }
}
```

### How did you run Stylelint?

CLI, VSCode official Stylelint extention and Demo on the official website.

### Which version of Stylelint are you using?

`^14.14.0`

### What did you expect to happen?

I expect `hoveringMenu` to be ignored when used as value in `getAppColor()`.

### What actually happened?

I get the error:
```shell
Expected "hoveringMenu" to be "hoveringmenu" (value-keyword-case)
```

On the line:
```scss
background-color: getAppColor(hoveringMenu, hover);
```

### Does the bug relate to non-standard syntax?

SCSS

### Proposal to fix the bug

I don't know how to fix the bug, but here are some tests I made:

These are working:
```js
getAppColor(hoveringMenu, background)
"ignoreFunctions": ["/getappcolor/"]

getAppColor(hoveringMenu, background)
"ignoreFunctions": ["/getAppColor/i"]

getappcolor(hoveringMenu, background);
"ignoreFunctions": ["getappcolor"]
```

These are not working:
```js
getAppColor(hoveringMenu, background)
"ignoreFunctions": ["/getAppColor/"]

Foo(hoveringMenu, background);
"ignoreFunctions": ["Foo"]
```

Seems to be a problem of case...

**Comment by ybiquitous** [comment]
@Arkellys Thanks for opening the issue and for using the template. We can reproduce the false positive bug on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYLIBsICd4oDEIIAKACwgDcYcBLAOwHMBZGegVwEoBuFAXwA69EABoQAZwCe9AC5IAHgnDjxo8BHoAzWoyXAhKFAJA52mGOOP599Q4eOUkmdjAC0AaxiSA7riiuwJHEYKxQAbWNsbxpjETQjEB16XBgCdnowGVoNSwRw4yIIYwBdPmKDfiE+ED4gA).

Perhaps, the following code may be related to the bug:

https://github.com/stylelint/stylelint/blob/7fbf2972a3c0d3d0d837fe2ba7964478fc323b7a/lib/rules/value-keyword-case/index.js#L108-L113

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
> Seems to be a problem of case...

Sounds likely.

The solution maybe as simple as:

```diff js
- optionsMatches(secondaryOptions, 'ignoreFunctions', valueLowerCase) 
+ optionsMatches(secondaryOptions, 'ignoreFunctions', node.value)
```

And adding a testcase.

Our `ignore*` options are typically case sensitive as a`/i` regex can be used for case insensitivity. It's a bug that this option doesn't follow this pattern. I assume it's never come up before as everyone who's currently using the option has lowercase function name.
